cmake/CompileBoost.cmake

```diff
@@ -6,7 +6,15 @@ function(compile_boost)
                           "${multiValueArgs}" ${ARGN} )
   # Configure the boost toolset to use
   set(BOOTSTRAP_ARGS "--with-libraries=context")
+
   set(B2_COMMAND "./b2")
+  set(B2_BUILD_ARGS "")
+
+  if (UNIX AND CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
+    set(BOOTSTRAP_ARGS ${BOOTSTRAP_ARGS} "--with-toolset=clang")
```

Comment:
I am pretty sure this will break on macOS. There you need to use the toolset `darwin`.

Reply:
I have verified that using `clang` would allow FDBServer to compile on Mac. No toolset `darwin` is needed. Not quite sure what is the root cause of the issue.